lib/proxy.js

```diff
@@ -174,7 +174,7 @@ function ReverseProxy(opts){
     if(target){
       proxy.ws(req, socket, head, {target: target});
     }else{
-      notFound(socket);
+      respondNotFound(req, socket);
```

Comment:
it has the write and the end, I am not sure of the status method, probably not. I think that this codepath never happens in practice though, thats why it has been working so far.